llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -11307,28 +11307,34 @@ static SDValue foldBoolSelectToLogic(SDNode *N, SelectionDAG &DAG) {
   if (VT != Cond.getValueType() || VT.getScalarSizeInBits() != 1)
     return SDValue();
 
-  // select Cond, Cond, F --> or Cond, F
-  // select Cond, 1, F    --> or Cond, F
+  auto FreezeIfNeeded = [&](SDValue V) {
+    if (!DAG.isGuaranteedNotToBePoison(V))
```

Comment:
getNode(ISD::FREEZE) should do this for us?
https://github.com/llvm/llvm-project/blob/02cb89b36a7ae9be4ab657306b69dc9d2830d0d5/llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp#L5718-L5723

Reply:
Ah. Nice!
Downstream I also needed some additional checks to just do this for my target (since it is a pain doing it for all targets until this has landed upstream). So that is one reason why I've wrapped the check this way.